python/test/unit/language/test_core.py

```diff
@@ -1987,6 +1987,36 @@ def kernel(I, O):
         kernel[(1, )](I, O)
 
 
+@pytest.mark.interpreter
+@pytest.mark.parametrize("dtype_str", ["int32", "float16"])
+@pytest.mark.parametrize("size", [1, 4, 16])
+@pytest.mark.parametrize("op", ["add", "cas"])
+def test_tensor_atomic_use_result(dtype_str, size, op, device):
+    if is_hip():
+        pytest.skip(
```

Comment:
@scxiao @antiagainst fixing these problems may take a while for me. If you guys have any available time feel free to take it. Otherwise we will just skip the tests for now.